tests/std/tests/P0053R7_cpp_synchronized_buffered_ostream/test.cpp

```diff
@@ -0,0 +1,285 @@
+                      && is_same_v<typename allocator_traits<Alloc>::is_always_equal, true_type>) {
+            assert(buf1.get_allocator() == buf2.get_allocator());
+        } else if constexpr (is_same_v<typename allocator_traits<Alloc>::is_always_equal, true_type>) {
+            assert(buf1.get_allocator() == buf2.get_allocator());
+        } else {
+            assert(buf1.get_allocator() != buf2.get_allocator());
+        }
+    }
+    assert(buf->str == "Some input");
+    buf->str.clear();
+    { // test swap
+        Syncbuf buf1{buf};
+        OStream{&buf1} << "Some input that requires re-allocation";
+        Syncbuf buf2{nullptr};
+        auto buf1WrappedObject = buf1.get_wrapped();
+        auto buf1BufferSize    = buf1.Test_get_buffer_size();
+        auto buf1DataSize      = buf1.Test_get_data_size();
+        auto buf2WrappedObject = buf2.get_wrapped();
+        auto buf2BufferSize    = buf2.Test_get_buffer_size();
+        auto buf2DataSize      = buf2.Test_get_data_size();
+        if constexpr (allocator_traits<Alloc>::propagate_on_container_swap::value
+                      || is_same_v<typename allocator_traits<Alloc>::is_always_equal, true_type>) {
+            buf1.swap(buf2);
+            assert(buf2.get_wrapped() == buf1WrappedObject);
+            assert(buf2.Test_get_buffer_size() == buf1BufferSize);
+            assert(buf2.Test_get_data_size() == buf1DataSize);
+            assert(buf1.get_wrapped() == buf2WrappedObject);
+            assert(buf1.Test_get_buffer_size() == buf2BufferSize);
+            assert(buf1.Test_get_data_size() == buf2DataSize);
+        }
+    }
+    assert(buf->str == "Some input that requires re-allocation");
+    buf->str.clear();
+}
+
+int main() {
+    string_buffer<char> char_buffer{};
+    string_buffer<char, true> no_sync_char_buffer{};
+
+    test_syncbuf_member_functions<allocator<char>>();
+    test_syncbuf_member_functions<small_size_allocator<char>>();
+    test_syncbuf_member_functions<fancy_ptr_allocator<char>>();
+    test_syncbuf_member_functions<small_size_fancy_ptr_allocator<char>>();
+
+    test_syncbuf_member_functions<small_size_allocator<char>>(&char_buffer);
+    test_syncbuf_member_functions<allocator<char>>(&char_buffer);
+    test_syncbuf_member_functions<fancy_ptr_allocator<char>>(&char_buffer);
+    test_syncbuf_member_functions<small_size_fancy_ptr_allocator<char>>(&char_buffer);
+
+    test_syncbuf_synchronization<allocator<char>>(&char_buffer);
+    test_syncbuf_synchronization<allocator<char>>(&no_sync_char_buffer);
+    test_syncbuf_synchronization<fancy_ptr_allocator<char>>(&char_buffer);
+    test_syncbuf_synchronization<fancy_ptr_allocator<char>>(&no_sync_char_buffer);
+
+    test_syncbuf_move_swap_operations<allocator<char>>(&char_buffer);
+    test_syncbuf_move_swap_operations<fancy_ptr_allocator<char>>(&char_buffer);
+    // test_syncbuf_move_swap_operations<non_move_assignable_non_equal_allocator<char>>(&char_buffer);

```

Comment:
This test is failing with `Exit Code: 3221225477`, and it's also failing locally on my local machine with MSVC.
It fails with an exception while exiting the destructor of `buf1` object in the ```// test move assignment``` section with access violation from msvcp140d.dll. I am not sure how to get to the bottom of this one, any help 😕?